core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -193,8 +217,19 @@ private[spark] object HighlyCompressedMapStatus {
     } else {
       0
     }
+    val threshold = SparkEnv.get.conf.get(config.SHUFFLE_ACCURATE_BLOCK_THRESHOLD)
+    val hugeBlockSizesArray = ArrayBuffer[Tuple2[Int, Byte]]()
+    if (numNonEmptyBlocks > 0) {
+      uncompressedSizes.zipWithIndex.foreach {
```

Comment:
The existing code tried to optimize for performance by using an imperative while loop instead of Scala collections operations. Thus I don't think we should be using `zipWithIndex` and `foreach` here. In addition, it would be better to do this during a single loop rather than two loops.